app/src/main/java/com/eveningoutpost/dexdrip/services/Ob1G5CollectionService.java

```diff
@@ -2167,15 +2167,22 @@ public void run() {
                 if (!battery_status.equals("OK"))
                     l.add(new StatusItem("Transmitter Status", battery_status, BAD));
             }
-            Highlight TX_dys_highlight; // Transmitter Days highlight
+            Highlight TX_dys_highlight = NORMAL; // Transmitter Days highlight
             final int TX_dys = DexTimeKeeper.getTransmitterAgeInDays(tx_id); // Transmitter days
-            if (vr != null && (FirmwareCapability.isTransmitterRawCapable(getTransmitterID())) || TX_dys < 69) {
-                // Transmitter days < 69 or G5 or old G6 or modified Firefly
-                TX_dys_highlight = NORMAL;
-            } else if (TX_dys < 100) { // Unmodified Firefly with 68 < Transmitter days < 100
-                TX_dys_highlight = NOTICE;
-            } else { // Unmodified Firefly with transmitter days > 99
-                TX_dys_highlight = BAD;
+            if (vr != null && FirmwareCapability.isTransmitterModified(getTransmitterID())) { // Modified Firefly
```

Comment:
IMHO a more readable version (because of fewer conditions):
```java
            if (vr != null) {
                if (FirmwareCapability.isTransmitterModified(getTransmitterID())) { // Modified Firefly
                    if (TX_dys > 179) { // Transmitter days 180+
                        TX_dys_highlight = BAD;
                    } else if (TX_dys > 148) { // Transmitter days > 148
                        TX_dys_highlight = NOTICE;
                    }
                } else if (FirmwareCapability.isTransmitterRawIncapable(getTransmitterID())) { // Unmodified Firefly
                    if (TX_dys > 99) { // Transmitter days 100+
                        TX_dys_highlight = BAD;
                    } else if (TX_dys > 68) { // Transmitter days > 68
                        TX_dys_highlight = NOTICE;
                    }
                }
```